### PR TITLE
[12.0][ADD] sale_order_tag

### PR DESCRIPTION
**Pull request by patrickrwilson**
This module adds a tag to the sales order, not to be mistaken from the tag that already exists that is related to the opportunity. This tag is specific to sales orders.

### CODE REVIEW

File: sale_order_tag/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2020 Patrick Wilson <patrickraymondwilson@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': "Sale Order Tags",
+    'summary': """Adds Tags to Sales Orders.""",
+    'author': "Patrick Wilson, Odoo Community Association (OCA)",
+    'website': "https://github.com/OCA/project",
```
Comment (bodedra) [inline sale_order_tag/__manifest__.py]:
```suggestion
    'website': "https://github.com/OCA/sale-workflow",
```

**Review by bodedra — COMMENTED** [review]
minor suggestion

File: sale_order_tag/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2019 Patrick Wilson <patrickraymondwilson@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    so_tag_ids = fields.Many2many('sale.order.tag', string="Tags")
```
Comment (bodedra) [inline sale_order_tag/models/sale_order.py]:
```suggestion
    so_tag_ids = fields.Many2many('sale.order.tag', string="Sale Order Tags")
```

**Review by bodedra — COMMENTED** [review]
Function reviewed. Working great!

In sale order, Tags field labels already exist. So it will be confusing in import/export data to choose right field.

**Comment by patrickrwilson** [comment]
@bodedra Thanks for your review and suggestions, I've committed both.

**Comment by bodedra** [comment]
Looks good. Let's wait for CI :green_apple:

**Review by HaraldPanten — APPROVED** [review]
I don't know why runbot is failing. It seems to be working fine. Functional review 👍

**Comment by bodedra** [comment]
There is an issue inside readme.

Screenshot from 2020-09-25 10-06-35

File: sale_order_tag/models/sale_order_tag.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Copyright (C) 2019 Pavlov Media
+# License Proprietary. Do not copy, share nor distribute.
+
+from odoo import fields, models
+
+
+class SaleOrderTags(models.Model):
```
Comment (legalsylvain) [inline sale_order_tag/models/sale_order_tag.py]: Should be singular.

File: sale_order_tag/views/sale_order_tag.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<!-- Copyright 2019 Pavlov Media
+     # License Proprietary. Do not copy, share nor distribute. -->
+
+<odoo>
+    <record id="sale_order_tag_seq" model="ir.sequence">
```
Comment (legalsylvain) [inline sale_order_tag/views/sale_order_tag.xml]: This records should be in another file, in a data folder with noupdate to 1.

File: sale_order_tag/views/sale_order_tag.xml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            </tree>
+        </field>
+    </record>
+
+    <record id="sale_order_tag_form_view" model="ir.ui.view">
+        <field name="name">sale.order.tag.form</field>
+        <field name="model">sale.order.tag</field>
+        <field name="arch" type="xml">
+            <form string="Sale Order Tags">
+                <sheet>
```
Comment (legalsylvain) [inline sale_order_tag/views/sale_order_tag.xml:24]: Could you add a smar button archive, with the active field ?

File: sale_order_tag/views/sale_order.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_sales_order_filter"/>
+        <field name="arch" type="xml">
+            <field name='order_line' position="after">
+                <field name="so_tag_ids"/>
+            </field>
+        </field>
+    </record>
+
+    <menuitem name="Tags"
```
Comment (legalsylvain) [inline sale_order_tag/views/sale_order.xml]: This item should be better in the sale_order_tag.xml file.

File: sale_order_tag/models/sale_order_tag.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    _name = "sale.order.tag"
+    _order = "name"
+    _description = "Sale Order Tag"
+
+    name = fields.Char(required=True)
+    description = fields.Text(string="Description")
+    color = fields.Integer(string="Color Index")
+    team_ids = fields.Many2many('crm.team', string="Sales Teams")
+    active = fields.Boolean(string="Active", default=True)
+    sequence = fields.Integer(
```
Comment (legalsylvain) [inline sale_order_tag/models/sale_order_tag.py:18]: Could you add a company fields, optional to make this module working in a multi company context ?
(With the according ir.rule file)

**Review by legalsylvain — COMMENTED** [review]
Some minor remarks. LGTM otherwise.

**Comment by patrickrwilson** [comment]
@legalsylvain Thanks for your review, I believe I got all of your suggestions.

**Review by bodedra — APPROVED** [review]

**Review by legalsylvain — APPROVED** [review]
thanks a lot ! 

(code review / no test).

**Comment by rousseldenis** [comment]
@patrickrwilson If this is ready, could you squash your commits ?

**Comment by patrickrwilson** [comment]
@rousseldenis Yes, it's ready, I just squashed it. Thanks everyone.

**Pull request merged**